Automattic/jetpack
# Support unparsed apiFetch responses for VideoPress uploads

An issue came up on one particular customer's blogs where VideoPress the final UI update after a video upload was performed from the block editor would fail after the upload completed and a network request to `/wp/v2/media/undefined` was seen in the Network console.

This issue was narrowed down to some `apiFetch` middleware that was being added by another plugin
<img width="589" alt="Screen Shot 2021-10-13 at 4 35 33 PM" src="https://user-images.githubusercontent.com/4081020/137209679-5de91a13-473e-4397-a587-d3cf227347b2.png">

This was causing the VideoPress upload callback code to fail because it was expecting a json object representing a media response, but instead it was receiving a Response object. This PR adds support for both parsed and unparsed body responses.

Fixes #

#### Changes proposed in this Pull Request:

* This change adds support for parsing apiFetch response bodies if `options.parse` has been disabled by any other middleware loaded by other plugins.

#### Jetpack product discussion



#### Does this pull request change what data or activity we track or use?


nope

#### Testing instructions:






* Start with a Jetpack site with VideoPress enabled
* build jetpack `jetpack build plugins/jetpack`
* Add a `/video` block to a post and upload a video
* The upload should succeed and your video upload preview should be displayed
* Edit `projects/plugins/jetpack/modules/videopress/js/gutenberg-video-upload.js` and add `options.parse = false;` around  line 40
* build jetpack `jetpack build plugins/jetpack` again
* Refresh your post editor and add a video block and upload a video again
* The upload should succeed
* If you want to test the original issue, remove the changes from this PR but add back the `options.parse = false;` line, then rebuild jetpack plugin. You should see the UI flicker back to the upload block after attempting to upload a video and you will see the 404 for `/wp/v2/media/undefined` in the Network console.
